src/main/groovy/io/cify/framework/core/Capabilities.groovy

```diff
@@ -55,37 +74,78 @@ class Capabilities {
         }
-        throw new CifyFrameworkException("Unsupported device category $category")
+        return new DesiredCapabilities(result)
+    }
+
+    /**
+     * Returns desired capabilities for given capabilityId
+     *
+     * @param capabilityId capabilities identifier
+     * @return DesiredCapabilities
+     */
+    DesiredCapabilities toDesiredCapabilities(String capabilityId) {
+        LOG.debug(MARKER, "Get desired capabilities for capabilityID $capabilityId")
+        Map result = [:]
+        try {
+            result = capabilityList.find { it.id == capabilityId && it.available }.capabilities as LazyMap
+        } catch (NullPointerException) {
+            new CifyFrameworkException("No available capabilities for $capabilityId found")
+        }
+        return new DesiredCapabilities(result)
     }
 
     /**
      * Adds to desired capabilities
+     *
      * @param category - Device category
      * @param key - key to add
      * @param value - value to add
      */
-    void addToDesiredCapabilities(DeviceCategory category, String key, def value) {
+    void addToDesiredCapabilities(DeviceCategory category, String key, String value) {
         LOG.debug(MARKER, "Add to desired capabilities for category $category")
-        switch (category) {
-            case DeviceCategory.BROWSER:
-                browser.put(key, value)
-                break
-            case DeviceCategory.ANDROID:
-                android.put(key, value)
-                break
-            case DeviceCategory.IOS:
-                ios.put(key, value)
-                break
-            default:
-                throw new CifyFrameworkException("Unsupported device category $category")
+        try {
+            capabilityList.findAll { (it.category as String).toUpperCase() == category.toString() }.each {
+                it.capabilities.put(key, value)
+            }
+        } catch (NullPointerException) {
+            throw new CifyFrameworkException("Could not add $key:$value. Capabilities for category $category not found.")
+        }
+    }
+
+    /**
+     * Sets available key of capability with given capabilityId to given value
+     *
+     * @param capabilityId id of the capability
+     *
+     * @param value value given to available key
+     */
+    void setAvailable(String capabilityId, boolean isAvailable) {
+        try {
+            capabilityList.find { it.id == capabilityId }.available = isAvailable
+        } catch (NullPointerException) {
+            throw new CifyFrameworkException("Setting \"available\":$isAvailable failed. Capability with capabilityId:$capabilityId not found")
+
+        }
+    }
+
+    /**
+     * Gets category from capability with given id
+     * @param capabilityId id of the capability
+     * @return
+     */
+    DeviceCategory getCategory(String capabilityId) {
+        try {
+            DeviceCategory.valueOf((capabilityList.find { it.id == capabilityId }.category as String).toUpperCase())
```

Comment:
is it returned value?

Reply:
No, DeviceCategory is returned value, while 'it' is a Map, representing capability.